src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/sendorder/SendOrderPartyDocumentHandler.java

```diff
@@ -122,13 +122,17 @@ private void addAdditionalOrderDocumentToPartyCollection(FinremCaseData caseData
     protected boolean shouldAddDocumentToOrderColl(CaseDocument document,
                                                    List<ApprovedOrderConsolidateCollection> orderCollForRole) {
         List<ApprovedOrderConsolidateCollection> existingCollection = Optional.ofNullable(orderCollForRole)
-                .orElse(new ArrayList<>());
+            .orElse(new ArrayList<>());
         if (existingCollection.isEmpty()) {
             return true;
         }
-        return existingCollection.stream().noneMatch(doc -> doc.getValue().getApproveOrders().stream().anyMatch(order ->
-                order.getValue().getCaseDocument().getDocumentFilename().equals(ADDITIONAL_HEARING_FILE_NAME)
-                        && order.getValue().getCaseDocument().getDocumentUrl().equals(document.getDocumentUrl())
+        return existingCollection.stream().noneMatch(doc -> doc.getValue().getApproveOrders().stream().anyMatch(order -> {
+                if (order.getValue().getCaseDocument() != null) {
```

Comment:
Could you extract this if clause to a method to make it a bit cleaner to follow?

Reply:
made change